### PR TITLE
[exporter/elasticsearch] Add exponential histogram support

### LINKED ISSUE #34813: [exporter/elasticsearch] Support exponential histogram
### Component(s)

exporter/elasticsearch

### Is your feature request related to a problem? Please describe.

There is no exponential histogram support in elasticsearch exporter now. It means exponential histograms are ignored. As Elasticsearch does not support native exponential histogram at the moment, exponential histograms in opentelemetry should be converted to T-digest and indexed.

### Describe the solution you'd like

Convert to exponential histograms to Elasticsearch compatible format and index them.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by carsonip**
**Description:** <Describe what has changed.>

Convert exponential histograms to T-digest histograms supported by Elasticsearch

**Link to tracking Issue:** <Issue number if applicable>

Fixes #34813

**Testing:** <Describe what testing was performed and which tests were added.>

Unit tests and exporter test

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/elasticsearchexporter/model.go [hunk 1]
```diff
@@ -480,6 +481,52 @@ func summaryToValue(dp pmetric.SummaryDataPoint) pcommon.Value {
...
+		}
+		lb := -exphistogram.LowerBoundary(offset+i+1, scale)
+		ub := -exphistogram.LowerBoundary(offset+i, scale)
+		counts.AppendEmpty().SetInt(int64(count))
+		values.AppendEmpty().SetDouble(lb + (ub-lb)/2)
+	}
+
+	if zeroCount := dp.ZeroCount(); zeroCount != 0 {
+		counts.AppendEmpty().SetInt(int64(zeroCount))
+		// FIXME: do we really have to take the median of last negative and first positive, instead of using 0?
```
Comment (felixbarny) [inline exporter/elasticsearchexporter/model.go:508]: I'd expect us to be able to just use `0`. If the offsets if the positive and negative scale are the same, the midpoint is always zero. If the offsets are different, I don't think the midpoint value is meaningful. As the zero count is the count of values that are zero (or within the `zero_threshold`), I think we should just use `0` as the value`.

**Review by axw — APPROVED** [review]
Looks great!

File: exporter/elasticsearchexporter/internal/exphistogram/exphistogram.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		ub := -LowerBoundary(offset+i, scale)
+		counts = append(counts, int64(count))
+		values = append(values, lb+(ub-lb)/2)
+	}
+
+	if zeroCount := dp.ZeroCount(); zeroCount != 0 {
+		counts = append(counts, int64(zeroCount))
+		// The mean centroid is only non-zero when positive offset and negative offset are not the same,
+		// but the mean centroid between negative and positive boundaries closest to zero will not be very meaningful anyway.
+		// Using a zero here instead.
```
Comment (felixbarny) [inline exporter/elasticsearchexporter/internal/exphistogram/exphistogram.go:52]: For zero counts, calculating centroids are not really relevant. We know the value is zero, and it doesn't depend on the negative or positive offset. Maybe consider clarifying or just removing the comment.

**Review by felixbarny — APPROVED** [review]

File: exporter/elasticsearchexporter/internal/exphistogram/exphistogram.go (hunk 2 above)
Comment (carsonip) [reply exporter/elasticsearchexporter/internal/exphistogram/exphistogram.go:52]: Right. I was under the false impression that anything that falls between first negative and positive boundaries from zero are in the zero bucket. It is wrong. I believe the producer should adjust offset and scale to accommodate such values. Zero bucket really store values around zero.

**Review by lahsivjar — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**